ndcube/utils/wcs.py

```diff
@@ -209,13 +209,18 @@ def _wcs_slicer(wcs, missing_axis, item):
     # what was inputed so we had a reverse missing_axis.
     extra_coords_list = [] # initiating new list
     for slice_element in item_: # check item_ slices for dropped axes if any
-        if (slice_element.stop - slice_element.start) == 1:
-           extra_coords_list.append(slice_element)
-        else:
-           pass
+        for _bool in missing_axis:
+            if not _bool:
+                if index is not 1: 
+                    if (slice_element.stop - slice_element.start) == 1:
+                        px, py = slice_element.all_pix2world(slice_element.wcs.crpix[0], slice_element.wcs.crpix[1], 1)
```

Comment:
The ```slice``` element, ```slice_element```, doesn't have the ```all_pix2world``` method.  The ```wcs``` object has is.  So we need to call this method and supply the pixel coordinates using the ```slice_element```.
```python
pix_coords = [0] * len(item_) # Set up a list of pixel coords as input to all_pix2world
pix_coords[i] = slice_element.start # Enter pixel coordinate for this axis.  Since we are only dealing with independent axes the other axes can remain at 0.
real_world_coords = wcs.all_pix2world(tuple(pix_coords), 0)
```

Reply:
To learn more about ```astropy.wcs.WCS.all_pix2world```, see the docs: http://docs.astropy.org/en/stable/api/astropy.wcs.WCS.html#astropy.wcs.WCS.all_pix2world

Try playing around with it a bit to understand how it works.  This should help you in this PR.

Reply:
Yup @DanRyanIrish, I definitely should have been more careful with **not** treating the `slice_element` as an `wcs` object.